helix-term/src/commands/lsp.rs

```diff
@@ -304,10 +304,14 @@ fn diag_picker(
                   offset_encoding,
               },
               action| {
-            jump_to_position(cx.editor, path, diag.range, *offset_encoding, action)
+            jump_to_position(cx.editor, path, diag.range, *offset_encoding, action);
+            let (view, doc) = current!(cx.editor);
+            view.diagnostics_handler
+                .immidietly_show_diagnostic(doc, view.id);
         },
     )
     .with_preview(move |_editor, PickerDiagnostic { path, diag, .. }| {
+        // TODO: render this single diagnostic inline?
```

Comment:
Being able to attach decorations (virtual text, diagnostic highlights, etc.) in the preview could be useful. It could also be nice to have the diff gutter in git change previews, so we may want to consider larger changes to the preview (separate work from this or #9647 though, just something to keep in mind)

Reply:
yeah I think we can do more with the preview but definitely deferred. I sthould probably just remove this todo to avoid conflicts.